app/resources/game_content.py

```diff
@@ -0,0 +1,203 @@
+    ram: List[str] = elements["ram"]
+    gpu: str = elements["gpu"]
+    disk: List[str] = elements["disk"]
+
+    performance_cpu: float = hardware["cpu"][cpu]["cores"] * hardware["cpu"][cpu]["frequencyMax"]
+
+    performance_ram: float = 1
+    for ram_stick in ram:
+        performance_ram += (
+            min(
+                resolve_ram_type[hardware["mainboards"][motherboard]["ram"]["type"]],
+                resolve_ram_type[hardware["ram"][ram_stick]["ramTyp"]],
+            )
+            * hardware["ram"][ram_stick]["ramsize"]
+        )
+
+    performance_gpu: float = resolve_gpu_type[hardware["graphiccards"][gpu]["interface"]] * math.sqrt(
+        hardware["graphiccard"][gpu]["ramsize"] * hardware["graphiccard"][gpu]["frequency"]
+    )
+
+    dick_storage: float = 1
+
+    for i in disk:
+        dick_storage += hardware["diskStorage"][i]["capacity"] * math.log1p(
+            hardware["diskStorage"][i]["writingSpeed"] * hardware["diskStorage"][i]["readingSpeed"]
+        )
+
+    network: float = hardware["mainboards"][motherboard]["speed"]
+
+    return performance_cpu, performance_ram, performance_gpu, dick_storage, network
+
+
+def add_able(elements: dict, type_new: str) -> bool:
+    if type_new == "cpu" and "cpu" in elements.keys():
+        return False
+    if type_new == "motherboard" and "motherboard" in elements.keys():
+        return False
+    if type_new == "gpu" and "gpu" in elements.keys():
+        return False
+    if type_new == "harddrive":
+        # TODO check for multiple harddrives dependinng on spaces in the motherboard
+        pass
+
+    return True
+
+
+def create_hardware(elements: dict, device_uuid: str) -> None:
+    Hardware.create(device_uuid, elements["cpu"], "cpu")
+    Hardware.create(device_uuid, elements["gpu"], "gpu")
+    Hardware.create(device_uuid, elements["motherboard"], "mainboard")
+    for disk in elements["disk"]:
+        Hardware.create(device_uuid, disk, "diskStorage")
+    for ram in elements["ram"]:
+        Hardware.create(device_uuid, ram, "ram")
+
+
+def scale_resources(s: List[Service], scale: Tuple[float, float, float, float, float]):
+    for service in s:
+        send: dict = {
+            "service_uuid": service.service_uuid,
+            "cpu": scale[0] * service.allocated_cpu,
+            "ram": scale[1] * service.allocated_raml,
+            "gpu": scale[2] * service.allocated_gpu,
+            "disk": scale[3] * service.allocated_disk,
+            "network": scale[4] * service.allocated_network,
+        }
+
+        m.contact_microservice("service", ["hardware", "scale"], send)
+
+
+def generate_scale(
+    data: Tuple[float, float, float, float, float], wl: Workload
+) -> Tuple[float, float, float, float, float]:
+    if wl.usage_cpu + data[0] < wl.performance_cpu:
+        cpu: float = 1
+    else:
+        scale: float = (1 - (wl.usage_cpu + data[0] - wl.performance_cpu)) / (wl.usage_cpu + data[0])
+        cpu: float = scale
```

Comment:
simplify this

Reply:
you can not simplify this math term ... they all follow the same equation so it should be fine

Reply:
But `scale` is not used anymore. So you could write the result of the equation directly to `cpu`

Reply:
And how did you get this equation? Because I think
```python
cpu: float = wl.performance_cpu / (wl.usage_cpu + data[0])
```
should be the correct way to calculate the scale factor

Reply:
We can use your equation ... I thought again too complex ... Your should also be more resource saving